eng/pipelines/templates/steps/build.yml

```diff
@@ -0,0 +1,32 @@
+parameters:
```

Comment:
With the addition of this file I wonder if we can remove the build and vet steps from `build-test.yml` (probably rename it to simply `test.yml`), then apply both templates in `archetype-sdk-client.yml`.  This way we don't duplicate the content.

@scbedd for his input.

Reply:
I did think about doing the same. I'll make the change if Scott thinks it's fine.